### PR TITLE
Fixes #24572 - wrap collection id in array

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: app/models/concerns/dirty_associations.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@ def dirty_has_many_associations(*args)
           #   end
           define_method "#{association_ids}=" do |collection|
             collection ||= [] # in API, #{association}_ids is converted to nil if user sent empty array
+            collection = [collection].flatten # in case we got single id, we ensure it's an array
```
Comment (adamruzicka) [inline app/models/concerns/dirty_associations.rb]: Using `Array(collection)` might be faster

Comment (ares) [reply app/models/concerns/dirty_associations.rb]: updated

**Review by adamruzicka — DISMISSED** [review]
Fixes the issue for me

File: app/models/concerns/dirty_associations.rb [hunk 2]
```diff
@@ -33,6 +33,7 @@ def dirty_has_many_associations(*args)
           #   end
           define_method "#{association_ids}=" do |collection|
             collection ||= [] # in API, #{association}_ids is converted to nil if user sent empty array
+            collection = Array(collection) # in case we got single id, we ensure it's an array
```
Comment (tbrisker) [inline app/models/concerns/dirty_associations.rb]: This makes the previous line redundant:
```ruby
> Array(nil)
=> []

Comment (ares) [reply app/models/concerns/dirty_associations.rb]: updated

**Comment by tbrisker** [comment]
While this works, I'm concerned that we might be hiding away the root problem here, assigning a single number to the ids field might indicate something went wrong along the way (such as taking the first element of the array instead of the whole array)

**Comment by ares** [comment]
I looked at the AR behavior and to my surprise it works the same way. I don't know where it passes single id exactly, but this method should mimic AR behavior, so we need to support single id anyway.

**Review by tbrisker — APPROVED** [review]
Ok, in that case I'm fine with this, I just hope we aren't hiding away some bug. failed test unrelated. Thanks @ares !

**Comment by tbrisker** [comment]
cherry-picked to 1.19 as 45e76111d

**Pull request merged**